### PR TITLE
Hessian function refactor.

### PR DESCRIPTION
**Pull request by ByzanTine**
Majorly for readability.

### CODE REVIEW

File: cifar10-pytorch/optimizer_tr_cg.py [hunk 1]
```diff
@@ -104,57 +103,29 @@ def hess_fn(self, v, grad_graph_closure):
...
-        end = time.time()
-        print("time: " + str(end - start))
-        print("accum: " + str(accum))
-
-        old_accum = accum
-        start = time.time()
         flat_grad = torch.cat(([
             param_grad_nodes[idx].view(-1)
-            for idx, p in enumerate(self._params)
+            for idx, _ in enumerate(self._params)
```
Comment (montaguegabe) [inline cifar10-pytorch/optimizer_tr_cg.py]: this should probably just be `for idx in range(len(self._params))`

File: cifar10-pytorch/optimizer_tr_cg.py [hunk 2]
```diff
@@ -104,57 +103,29 @@ def hess_fn(self, v, grad_graph_closure):
...
         ]))
         accum = flat_grad @ v
```
Comment (montaguegabe) [inline cifar10-pytorch/optimizer_tr_cg.py]: This looks much much nicer than the way I was doing it before. However, definitely double check with pdb or print statements that at this line, there is gradient attached to flat_grad, and that none of your operations have destroyed the backwards trail in the computational graph.

Have you verified this?

File: cifar10-pytorch/util.py [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
 
 FORMAT = '[%(asctime)-15s %(filename)s:%(lineno)s] %(message)s'
 FORMAT_MINIMAL = '%(message)s'
 
 logger = logging.getLogger('trcg')
 logging.basicConfig(format=FORMAT)
 logger.setLevel(logging.DEBUG)
+
+
+def flatten(grad_calculator):
```
Comment (montaguegabe) [inline cifar10-pytorch/util.py]: This is very cool! I didn't know how to make these guys before 👍

**Comment by montaguegabe** [comment]
As long as you make sure that you haven't lost any of the gradient attachments in the computational graph in your new flattening procedures this looks good to merge to me

**Comment by montaguegabe** [comment]
Oh, although there was that minor thing where there is an "enumerate" call for no reason

**Comment by montaguegabe** [comment]
Oh, and perhaps we might want to consider applying the `@flatten` function decorator inside our optimizer, so that it is as easy to use for an outsider as possible. Minor though

File: cifar10-pytorch/optimizer_tr_cg.py (hunk 2 above)
Comment (ByzanTine) [reply cifar10-pytorch/optimizer_tr_cg.py]: Yes, I just checked the number of accum before&after the change, and all the steihaug calculated numbers. 

What do you mean by there should be a gradient attached to flat_grad? How do I check that?
